From patchwork Wed Oct 30 21:04:14 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Bill Wendling <morbo@google.com>
X-Patchwork-Id: 11220059
Return-Path: <SRS0=Bujf=YX=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 464DC913
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 30 Oct 2019 21:04:31 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id 1C11520862
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 30 Oct 2019 21:04:31 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (2048-bit key) header.d=google.com header.i=@google.com
 header.b="gXb60jE9"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727054AbfJ3VEa (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 30 Oct 2019 17:04:30 -0400
Received: from mail-pf1-f201.google.com ([209.85.210.201]:39253 "EHLO
        mail-pf1-f201.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726749AbfJ3VEa (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 30 Oct 2019 17:04:30 -0400
Received: by mail-pf1-f201.google.com with SMTP id l20so2696506pff.6
        for <kvm@vger.kernel.org>; Wed, 30 Oct 2019 14:04:29 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=date:in-reply-to:message-id:mime-version:references:subject:from:to
         :cc;
        bh=upjAEuFJdt1TLpNgpgQupJ9CCvCrQ2wU04Vj0IjMAHs=;
        b=gXb60jE9rGGecCIJUp0YW459p0D2wHyP/rhiOgFm1XwVu9qHHVDofMQ3kAvjwZGojt
         133ZLoNs5EhAuq7l3t9SumssAKOLIBkUNTjsplliseIog+g5BAF7/oJfQy1uDiBVyI6V
         53tEBdpjDOFEsH5qvu/EyW2lXtWWaedAX8wqDwZPbjQsKhFIlw7M/kby5cyTcr4VC0Wy
         KFUShzJVp3GdRkOKpGgzgl3jha8A7Jtr8fAN5LVrHu+VEZWXn5+yT3V1tdrGkR2wykCd
         22Zm13a8sSTpzLoYSrTYkXNPmoISoY5jtIn9GCb3aBfM4cU3N8KW0GJ6BqWgLffIunVK
         yO/g==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:date:in-reply-to:message-id:mime-version
         :references:subject:from:to:cc;
        bh=upjAEuFJdt1TLpNgpgQupJ9CCvCrQ2wU04Vj0IjMAHs=;
        b=luPHG9HHe/zX5DogU3Mh6H9cHUpe5Ph2w6PRFEFeEY98lAfKVjTqNGxvasBuQBXLyt
         +FM5E2aRX0oMfRoF6QJkOBlnixyInwgDCjENZ/ltExd2kAmsvqvH5rzDFBfhQP0I5gmf
         R5O46V+NHOSw9HFNdvqw7Tk8mMjQcyA+qGrFO1wyroPM4jc+zsAQFc8otOkkyVZVgvjj
         T8yOWTTHzZBWUW4CFr/ptoX1ocNEw/qiD2VTaI3N2pnYARmMVv7QeI0V5W2KqxFIQId7
         gJIOHYwtRsWR+5SACQTm5SqkOXZfQx22XR743CjGMTJyDVDNqJdiMBYUo9MYHkTWmvEi
         leHQ==
X-Gm-Message-State: APjAAAVxjR+G1iAcyKtfwJhhkFen4Gj93cp0P122FQzGlY3Nbf9lrdGo
        hVf5EZ0O1n/aEONrf3bjk78DrSR1gRQtY55HC7i+vvsXN3obNkUZK3XI1RjmhlEgaBWdhKVzJxT
        gcHoGiaH2WsX7gbah2AgCBbf+qjeCiC6MQ6lcSOBk8Mk9gbRdrm8jXQ==
X-Google-Smtp-Source: 
 APXvYqz6psyX3VnEaTeKsc1C9it70gfs7BJaF/5WPERV1ZPXP7aD1wIeieT5JBHtPkGK8fDvKQMfwM3QkA==
X-Received: by 2002:a65:554e:: with SMTP id t14mr1606165pgr.370.1572469468375;
 Wed, 30 Oct 2019 14:04:28 -0700 (PDT)
Date: Wed, 30 Oct 2019 14:04:14 -0700
In-Reply-To: <20191030210419.213407-1-morbo@google.com>
Message-Id: <20191030210419.213407-2-morbo@google.com>
Mime-Version: 1.0
References: <20191015000411.59740-1-morbo@google.com>
 <20191030210419.213407-1-morbo@google.com>
X-Mailer: git-send-email 2.24.0.rc1.363.gb1bccd3e3d-goog
Subject: [kvm-unit-tests PATCH v3 1/6] x86: emulator: use "SSE2" for the
 target
From: Bill Wendling <morbo@google.com>
To: kvm@vger.kernel.org, pbonzini@redhat.com, thuth@redhat.com
Cc: jmattson@google.com, sean.j.christopherson@intel.com,
        Bill Wendling <morbo@google.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

The movdqu and movapd instructions are SSE2 instructions. Clang
interprets the __attribute__((target("sse"))) as allowing SSE only
instructions. Using SSE2 instructions cause an error.

Signed-off-by: Bill Wendling <morbo@google.com>
---
 x86/emulator.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x86/emulator.c b/x86/emulator.c
index 621caf9..bec0154 100644
--- a/x86/emulator.c
+++ b/x86/emulator.c
@@ -657,7 +657,7 @@ static bool sseeq(sse_union *v1, sse_union *v2)
     return ok;
 }
 
-static __attribute__((target("sse"))) void test_sse(sse_union *mem)
+static __attribute__((target("sse2"))) void test_sse(sse_union *mem)
 {
     sse_union v;
 

From patchwork Wed Oct 30 21:04:15 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Bill Wendling <morbo@google.com>
X-Patchwork-Id: 11220061
Return-Path: <SRS0=Bujf=YX=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id A2176913
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 30 Oct 2019 21:04:33 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id 75A0B20856
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 30 Oct 2019 21:04:33 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (2048-bit key) header.d=google.com header.i=@google.com
 header.b="MN27Cybj"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727074AbfJ3VEc (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 30 Oct 2019 17:04:32 -0400
Received: from mail-pf1-f202.google.com ([209.85.210.202]:48234 "EHLO
        mail-pf1-f202.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1727031AbfJ3VEc (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 30 Oct 2019 17:04:32 -0400
Received: by mail-pf1-f202.google.com with SMTP id g186so2687265pfb.15
        for <kvm@vger.kernel.org>; Wed, 30 Oct 2019 14:04:31 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=date:in-reply-to:message-id:mime-version:references:subject:from:to
         :cc;
        bh=oi3SVXxpNNUXh6Hxdlt6eW5Qo2+MN7VHWFvLbxNDRLg=;
        b=MN27Cybj0TFsMMAvzh/FIpsFd3kn7QvFkRdwmalSoTOQlX6Sxu792yu8p+aw0yGazi
         yLUtmAkDo6mGbGKLXNSZ1xjOgOvfEfRK3UwhLQjIYYPwqctLS6mFB8WX/3kej8S5pASm
         K/J0LvthOimNPAdOtiwaAzI+6+StYRJLAiP4I8hsAH94nn4mpqKRLRTQP1F3XfdC1Sen
         4nhN1kmukz+TmbSi4bpzN/RU4NddN3GMZuy+PJ/pOshV+EFmEXs/ZsjoF9wJlfoaTFkO
         D6MntIgq/0JVX/UUoVd1H8eWBpj0Xhs8LqY7ldS+gvuCUkOwB24llrCAnczQi24qN5f0
         UigA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:date:in-reply-to:message-id:mime-version
         :references:subject:from:to:cc;
        bh=oi3SVXxpNNUXh6Hxdlt6eW5Qo2+MN7VHWFvLbxNDRLg=;
        b=sm/4brldHDGqu05PRWHTAoa/LhbCSaVnUEShjGghPiQvhc+wL7oajOdoJEv/fsGTqB
         ykPEevRTRUarRgsQbzaH0CDQmFMiqPLIrkWiSwbfjluaLRTZot1yZy5lXzCn2+8lESWd
         DLkpIQVdYLmeYDrvsW3hh1PqL1iRcYxHb+QkhN6fsGI8+Xz8t4Hw7x0E2C4tXyMB4Hry
         kH5w4dSzETVBe+9Jh896Cli6sHV2pWjMf2bVp6H0FqS1XXMsc0/JVN24Zq8ImfUlqHTo
         3o2qIk5GXOsiEtuSQWzDgaIbftp96TF247hoTG356r1TC6kh6B9df6IBHXH/eOClMh54
         99AQ==
X-Gm-Message-State: APjAAAWgUVyZbvXyvXYsNcPYX+SPf/GT2NXLg8jimpqe5NiQCI9Vatj8
        GeOWhaKNygP/27N8uhYUj146ZSHpIGMJ0Gc/U04JIGYI6YDVN9b//0LCsM1KFJOOIlEr9AM+Ic2
        6UtlvSote627g7nua8en/EztdZaOK9jxxSb8ziHQZoWAIZbUW1ibB/Q==
X-Google-Smtp-Source: 
 APXvYqysvM9Sz8yl+1f4KEMKfJezgAPVxXrU+FNNNn8RgtKtPE/yunHgWrnzqzohEp2J6LqNVLd6AwrS6Q==
X-Received: by 2002:a63:234c:: with SMTP id u12mr1561512pgm.384.1572469471231;
 Wed, 30 Oct 2019 14:04:31 -0700 (PDT)
Date: Wed, 30 Oct 2019 14:04:15 -0700
In-Reply-To: <20191030210419.213407-1-morbo@google.com>
Message-Id: <20191030210419.213407-3-morbo@google.com>
Mime-Version: 1.0
References: <20191015000411.59740-1-morbo@google.com>
 <20191030210419.213407-1-morbo@google.com>
X-Mailer: git-send-email 2.24.0.rc1.363.gb1bccd3e3d-goog
Subject: [kvm-unit-tests PATCH v3 2/6] pci: cast the masks to the appropriate
 size
From: Bill Wendling <morbo@google.com>
To: kvm@vger.kernel.org, pbonzini@redhat.com, thuth@redhat.com
Cc: jmattson@google.com, sean.j.christopherson@intel.com,
        Bill Wendling <morbo@google.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

At this point, we're dealing with 32-bit addresses, therefore downcast
the masks to 32-bits.

Signed-off-by: Bill Wendling <morbo@google.com>
---
 lib/pci.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/pci.c b/lib/pci.c
index daa33e1..1b85411 100644
--- a/lib/pci.c
+++ b/lib/pci.c
@@ -107,7 +107,8 @@ pcidevaddr_t pci_find_dev(uint16_t vendor_id, uint16_t device_id)
 uint32_t pci_bar_mask(uint32_t bar)
 {
 	return (bar & PCI_BASE_ADDRESS_SPACE_IO) ?
-		PCI_BASE_ADDRESS_IO_MASK : PCI_BASE_ADDRESS_MEM_MASK;
+		(uint32_t)PCI_BASE_ADDRESS_IO_MASK :
+		(uint32_t)PCI_BASE_ADDRESS_MEM_MASK;
 }
 
 uint32_t pci_bar_get(struct pci_dev *dev, int bar_num)

From patchwork Wed Oct 30 21:04:16 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Bill Wendling <morbo@google.com>
X-Patchwork-Id: 11220063
Return-Path: <SRS0=Bujf=YX=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 49595913
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 30 Oct 2019 21:04:38 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id 2665D20873
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 30 Oct 2019 21:04:38 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (2048-bit key) header.d=google.com header.i=@google.com
 header.b="ThZ0KL4N"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727091AbfJ3VEh (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 30 Oct 2019 17:04:37 -0400
Received: from mail-pl1-f202.google.com ([209.85.214.202]:53170 "EHLO
        mail-pl1-f202.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1727079AbfJ3VEh (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 30 Oct 2019 17:04:37 -0400
Received: by mail-pl1-f202.google.com with SMTP id g4so1179891plj.19
        for <kvm@vger.kernel.org>; Wed, 30 Oct 2019 14:04:35 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=date:in-reply-to:message-id:mime-version:references:subject:from:to
         :cc;
        bh=BsEtptDhLoQ7dpINugZ/k0hkp7TfbimScSD2i2zd5S0=;
        b=ThZ0KL4N1ZvDkR86Lo1JPNSN5tVDlBU/YzyMTKCxX76ux/C7VuRlsn44X2VIJCLiPv
         aBS7Gkw+bxZrTM3spSqQGNoz7+WCEhJq513miltZKNg7XXNG7PEInYxsCs5YjFdnG0Wr
         QqG0BV6+G0x/rFhSa8bP87NyrQ3vOK92dsvnlv17ColLMJbvFcsxZDrQCehstY0w8h1q
         jYdgxnBdSEDcoUeUrhJo8RjsJmOhRUVx4W+rfqfe7wpf1cAgF/xMhPczQfEo0oLPXW04
         RftD2AsekxDkQ682nqy1IMx2I/FLsCfboE+NAwb7QVvYSidz37XcBX9t77nWRGr4NOHK
         QsGQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:date:in-reply-to:message-id:mime-version
         :references:subject:from:to:cc;
        bh=BsEtptDhLoQ7dpINugZ/k0hkp7TfbimScSD2i2zd5S0=;
        b=ZdgF068p+EmELUMvsZfZozqnxgjSbjORw3Jx8VmXHIIle5HMCKHS8R+/246Ygbqe99
         z7OQeKYkiY/rfIJnesl9wOv3j68hQFPILd1DmMJxvF0ErAaWMIRhgr0W5/dHaFO0Ybp3
         QQMVhquHWR2/3ljYxR/mGubpkx8pmBACGVdRqKFSYbJ7CoLrMIu2jtPuxGYs16vaiYGz
         Om4hAEr8Oz6arkHzRhPcfhtnjvyKx3wiACOFF8Q+A2dLskO0OkFuOdZjkBLEDxcStIyS
         xXysOH/z+nx1OezgwvFH5R5utFjom4yjWAcMnfWKvlEMfRtEw44WWmBaR7B3NIjm1OWH
         L2KA==
X-Gm-Message-State: APjAAAWIvUEegIVVWN5+fhdkBZgkzX/dOUU6g+OcvuwBkgixaYkejbNv
        TMNcPdeif6SBTvOV53TBEKx7eD+7EX2FlBEWwsgfjXiJtBpgeenbArebdvjI1OfPtyaO/i1lpCf
        JnLp74E9Elnf3IK4lRRmp4TVsqmyfxTyaFnDSMxDvlsCe3KRSN+q9pg==
X-Google-Smtp-Source: 
 APXvYqzG0Ax9LnXGFBRKa01rGs0a++gYZvn8FF9ElZIdeQqZXwXhzEZfWlvLe5iKXZk+KP8lwq1JufaNow==
X-Received: by 2002:a63:eb52:: with SMTP id b18mr1516544pgk.205.1572469474317;
 Wed, 30 Oct 2019 14:04:34 -0700 (PDT)
Date: Wed, 30 Oct 2019 14:04:16 -0700
In-Reply-To: <20191030210419.213407-1-morbo@google.com>
Message-Id: <20191030210419.213407-4-morbo@google.com>
Mime-Version: 1.0
References: <20191015000411.59740-1-morbo@google.com>
 <20191030210419.213407-1-morbo@google.com>
X-Mailer: git-send-email 2.24.0.rc1.363.gb1bccd3e3d-goog
Subject: [kvm-unit-tests PATCH v3 3/6] Makefile: use "-Werror" in cc-option
From: Bill Wendling <morbo@google.com>
To: kvm@vger.kernel.org, pbonzini@redhat.com, thuth@redhat.com
Cc: jmattson@google.com, sean.j.christopherson@intel.com,
        Bill Wendling <morbo@google.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

The "cc-option" macro should use "-Werror" to determine if a flag is
supported. Otherwise the test may not return a nonzero result. Also
conditionalize some of the warning flags which aren't supported by
clang.

Signed-off-by: Bill Wendling <morbo@google.com>
---
 Makefile | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/Makefile b/Makefile
index 32414dc..6201c45 100644
--- a/Makefile
+++ b/Makefile
@@ -46,13 +46,13 @@ include $(SRCDIR)/$(TEST_DIR)/Makefile
 # cc-option
 # Usage: OP_CFLAGS+=$(call cc-option, -falign-functions=0, -malign-functions=0)
 
-cc-option = $(shell if $(CC) $(1) -S -o /dev/null -xc /dev/null \
+cc-option = $(shell if $(CC) -Werror $(1) -S -o /dev/null -xc /dev/null \
               > /dev/null 2>&1; then echo "$(1)"; else echo "$(2)"; fi ;)
 
 COMMON_CFLAGS += -g $(autodepend-flags)
-COMMON_CFLAGS += -Wall -Wwrite-strings -Wclobbered -Wempty-body -Wuninitialized
-COMMON_CFLAGS += -Wignored-qualifiers -Wunused-but-set-parameter
-COMMON_CFLAGS += -Werror
+COMMON_CFLAGS += -Wall -Wwrite-strings -Wempty-body -Wuninitialized
+COMMON_CFLAGS += -Wignored-qualifiers -Werror
+
 frame-pointer-flag=-f$(if $(KEEP_FRAME_POINTER),no-,)omit-frame-pointer
 fomit_frame_pointer := $(call cc-option, $(frame-pointer-flag), "")
 fnostack_protector := $(call cc-option, -fno-stack-protector, "")
@@ -60,16 +60,20 @@ fnostack_protector_all := $(call cc-option, -fno-stack-protector-all, "")
 wno_frame_address := $(call cc-option, -Wno-frame-address, "")
 fno_pic := $(call cc-option, -fno-pic, "")
 no_pie := $(call cc-option, -no-pie, "")
+wclobbered := $(call cc-option, -Wclobbered, "")
+wunused_but_set_parameter := $(call cc-option, -Wunused-but-set-parameter, "")
+
 COMMON_CFLAGS += $(fomit_frame_pointer)
 COMMON_CFLAGS += $(fno_stack_protector)
 COMMON_CFLAGS += $(fno_stack_protector_all)
 COMMON_CFLAGS += $(wno_frame_address)
 COMMON_CFLAGS += $(if $(U32_LONG_FMT),-D__U32_LONG_FMT__,)
 COMMON_CFLAGS += $(fno_pic) $(no_pie)
+COMMON_CFLAGS += $(wclobbered)
+COMMON_CFLAGS += $(wunused_but_set_parameter)
 
 CFLAGS += $(COMMON_CFLAGS)
-CFLAGS += -Wmissing-parameter-type -Wold-style-declaration -Woverride-init
-CFLAGS += -Wmissing-prototypes -Wstrict-prototypes
+CFLAGS += -Woverride-init -Wmissing-prototypes -Wstrict-prototypes
 
 CXXFLAGS += $(COMMON_CFLAGS)
 

From patchwork Wed Oct 30 21:04:17 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Bill Wendling <morbo@google.com>
X-Patchwork-Id: 11220065
Return-Path: <SRS0=Bujf=YX=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id DC94013BD
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 30 Oct 2019 21:04:39 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id B9DF72087F
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 30 Oct 2019 21:04:39 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (2048-bit key) header.d=google.com header.i=@google.com
 header.b="AW992fJC"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727093AbfJ3VEi (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 30 Oct 2019 17:04:38 -0400
Received: from mail-qk1-f202.google.com ([209.85.222.202]:48512 "EHLO
        mail-qk1-f202.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1727099AbfJ3VEi (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 30 Oct 2019 17:04:38 -0400
Received: by mail-qk1-f202.google.com with SMTP id z64so3423923qkc.15
        for <kvm@vger.kernel.org>; Wed, 30 Oct 2019 14:04:37 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=date:in-reply-to:message-id:mime-version:references:subject:from:to
         :cc;
        bh=tP5VYyMFPyBsAqSM/e1ZfMjzucbp+c4GqFLPI5mcuDg=;
        b=AW992fJCZXBx956qeE2JdFQG1Vbfq3pWYaq+IPi3Lpg3GnG1Y7pF//+kjV7+NX+5wv
         t8eq4dBnSo4W2WGY3SOhICBGhR7E+gshJWwuB/qCYYIUvY+9m6yhACdODJLtIRlkwQQw
         oobIqJVsKs/8ujsx15RmPXd2gZEan2J0MgpsBs9bZUVFTyiUpfB9LYdk9gLm2j6B2ZBl
         zhbXCl6chkpJ4xc7DjivyJBd29wUk7sRL5mb4wxILJlnGWui8eC2ZzzIlFpwgGC0apmU
         bWE7JPxHbaziTsgmGlYASzh+4xdx6wWMdMQhFGYC28TvaZWakSaL05izMVLe+8MWymvO
         aK+A==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:date:in-reply-to:message-id:mime-version
         :references:subject:from:to:cc;
        bh=tP5VYyMFPyBsAqSM/e1ZfMjzucbp+c4GqFLPI5mcuDg=;
        b=UfTNGGV5UKkoWvSS2gk0TMc424yZuiSW3dBaujz1K14Ld2K2weetZvFx6Uxw0Wxrt/
         XK2wupNjqzshJs0FflCKW5i5DFqQQRrcYZ5HGv08dwhJ5n0ERmIdfT1VzE6flsYlNbeo
         nRin1zOfkR5MrXKrW566DyC32jw+/wJsgh4kvdiGO6oYK0UHsiSNcattkV7203RxxZpQ
         PUSvhrKcO3TBZ5x8ayzcA55LaWdK0Rd5QblBAOY2y2MGxSv3iQjath9DLeilHvkyrgpX
         fNl7WLUBLUTwdsXBomK/BLcZiCGilaD7Jl054KNXdkzAqrVo6OYqt/7I7Up8mq9NaRHO
         siuA==
X-Gm-Message-State: APjAAAWlZhzMIyLLY120FywPqa1wqcpR3xy5L2lDOUdJ2zGhjx+3s1/N
        OkOYttfi+yIysiRp0hcUu60lKApt3mOnKKcPKiPusZpXGvTIOT+tQa+q4BYmGlQ5AaQwH/+y/aP
        pw9hmVB94LRXVDUXJvH5+OgZO+Wo1O//lQP/Mt/Wcdu51HsnYJGqgpQ==
X-Google-Smtp-Source: 
 APXvYqyUJSCQqD3c0drhPQWOD2pFYe3h6vIeLr8FWx5hYto5aXXcfgL5N3jXcc6KZgh9Rv4dEHzZpd9GuQ==
X-Received: by 2002:a37:4ccb:: with SMTP id
 z194mr2044245qka.128.1572469476958;
 Wed, 30 Oct 2019 14:04:36 -0700 (PDT)
Date: Wed, 30 Oct 2019 14:04:17 -0700
In-Reply-To: <20191030210419.213407-1-morbo@google.com>
Message-Id: <20191030210419.213407-5-morbo@google.com>
Mime-Version: 1.0
References: <20191015000411.59740-1-morbo@google.com>
 <20191030210419.213407-1-morbo@google.com>
X-Mailer: git-send-email 2.24.0.rc1.363.gb1bccd3e3d-goog
Subject: [kvm-unit-tests PATCH v3 4/6] Makefile: add "cxx-option" for C++
 builds
From: Bill Wendling <morbo@google.com>
To: kvm@vger.kernel.org, pbonzini@redhat.com, thuth@redhat.com
Cc: jmattson@google.com, sean.j.christopherson@intel.com,
        Bill Wendling <morbo@google.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

The C++ compiler may not support all of the same flags as the C
compiler. Add a separate test for these flags.

Signed-off-by: Bill Wendling <morbo@google.com>
---
 Makefile | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 6201c45..9cb47e6 100644
--- a/Makefile
+++ b/Makefile
@@ -48,6 +48,8 @@ include $(SRCDIR)/$(TEST_DIR)/Makefile
 
 cc-option = $(shell if $(CC) -Werror $(1) -S -o /dev/null -xc /dev/null \
               > /dev/null 2>&1; then echo "$(1)"; else echo "$(2)"; fi ;)
+cxx-option = $(shell if $(CXX) -Werror $(1) -S -o /dev/null -xc++ /dev/null \
+              > /dev/null 2>&1; then echo "$(1)"; else echo "$(2)"; fi ;)
 
 COMMON_CFLAGS += -g $(autodepend-flags)
 COMMON_CFLAGS += -Wall -Wwrite-strings -Wempty-body -Wuninitialized
@@ -73,9 +75,19 @@ COMMON_CFLAGS += $(wclobbered)
 COMMON_CFLAGS += $(wunused_but_set_parameter)
 
 CFLAGS += $(COMMON_CFLAGS)
+CXXFLAGS += $(COMMON_CFLAGS)
+
+wmissing_parameter_type := $(call cc-option, -Wmissing-parameter-type, "")
+wold_style_declaration := $(call cc-option, -Wold-style-declaration, "")
+CFLAGS += $(wmissing_parameter_type)
+CFLAGS += $(wold_style_declaration)
 CFLAGS += -Woverride-init -Wmissing-prototypes -Wstrict-prototypes
 
-CXXFLAGS += $(COMMON_CFLAGS)
+# Clang's C++ compiler doesn't support some of the flags its C compiler does.
+wmissing_parameter_type := $(call cxx-option, -Wmissing-parameter-type, "")
+wold_style_declaration := $(call cxx-option, -Wold-style-declaration, "")
+CXXFLAGS += $(wmissing_parameter_type)
+CXXFLAGS += $(wold_style_declaration)
 
 autodepend-flags = -MMD -MF $(dir $*).$(notdir $*).d
 

From patchwork Wed Oct 30 21:04:18 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Bill Wendling <morbo@google.com>
X-Patchwork-Id: 11220067
Return-Path: <SRS0=Bujf=YX=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 97CD8913
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 30 Oct 2019 21:04:43 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id 7251820873
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 30 Oct 2019 21:04:43 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (2048-bit key) header.d=google.com header.i=@google.com
 header.b="m+cB/6qw"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727128AbfJ3VEm (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 30 Oct 2019 17:04:42 -0400
Received: from mail-pf1-f201.google.com ([209.85.210.201]:49380 "EHLO
        mail-pf1-f201.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1727102AbfJ3VEm (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 30 Oct 2019 17:04:42 -0400
Received: by mail-pf1-f201.google.com with SMTP id r187so2678555pfc.16
        for <kvm@vger.kernel.org>; Wed, 30 Oct 2019 14:04:40 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=date:in-reply-to:message-id:mime-version:references:subject:from:to
         :cc;
        bh=R5YsaR9bKKV9fNP9YXtzeLSXtxZmX8iRMZjNRdp+oWI=;
        b=m+cB/6qw1KFnvVJTUAVjF63cRK+2BS5naiUQ/mPU3UB1puPunOF7RogUGah/bOmqKf
         mWgFO/3eVEQatwHCUdsyAph8CD7LNxR9mzp/mKaMYe+z3OPNsGY2fH1TVXgDLrvVrMGa
         DG6SSaVzJ06IV8bv+CJ2ETTHsF/Zqo6FqNNt22gY4ZbV76mXon4sADKkjBtsWOlt9gZd
         Ygzh3o9zPwmOiMULDbxDS+2W5CHoSZAZqLrWKJOKR8/u9kq258EN6Vaa/4mASV5xsQhN
         yBfVUifHg5qHH/17yIO7Rq5SbEDsYBegSbxRty0yAHDlS39gXtKqiUPeF896BKbPO79Q
         EahA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:date:in-reply-to:message-id:mime-version
         :references:subject:from:to:cc;
        bh=R5YsaR9bKKV9fNP9YXtzeLSXtxZmX8iRMZjNRdp+oWI=;
        b=XErjjI+qJnt/Eo6IyAz4cX1UQ5a9FMrE3wE6GtG6/DKxrImJvmB9zJhQ9RVpsRV+yw
         RJOal53HLlkdpkHXMd+Qs9T/blURE9FHWsfuKmvjScZudWZdpDmGayN2tHef9dDsVugL
         DS7c9FNBumLHFrzFRPU6Id5AnXvAlQGHe8AtiDPAlaAW3q3Zcev+DsqlakokgnkJZLdM
         KXcj6Voe6U6R/wwjHzT6P8zHyX/QrWcj/6Fi2wc27NQ7QfzUf8ACe5RAHOp7qdzESs/7
         omU9EzZ6zhx8/ofVkTEO8TJ7Cz6sJKbSI7YPx/f0yVjUiwW0TGeH0M7LPX6HGQQ+Cdsh
         6VXA==
X-Gm-Message-State: APjAAAVkiC1v+S8k+pVkzEJnvr+fQmaDX0H1rE8RwThRAswbWync27oN
        +bYxp5QtHfY3CYvMsfyWcCTF8LMtY/jPbcR3TuULv0zql1MMLONotQ9iZS0Rfw1ojV5tgPg+KrJ
        P716VgSJgMwR3RnfTDP+4MSfTU7eXOHFa4eM4Y7cuDRDgqjkBiuFr7g==
X-Google-Smtp-Source: 
 APXvYqyF447WabAiYT/xTJmpjQsFAKB55cNNHE+8uaKTOBhkKbYeg72WGZF+ig55ZS5vp8vO+JQW4J5+gA==
X-Received: by 2002:a63:234c:: with SMTP id u12mr1562056pgm.384.1572469479458;
 Wed, 30 Oct 2019 14:04:39 -0700 (PDT)
Date: Wed, 30 Oct 2019 14:04:18 -0700
In-Reply-To: <20191030210419.213407-1-morbo@google.com>
Message-Id: <20191030210419.213407-6-morbo@google.com>
Mime-Version: 1.0
References: <20191015000411.59740-1-morbo@google.com>
 <20191030210419.213407-1-morbo@google.com>
X-Mailer: git-send-email 2.24.0.rc1.363.gb1bccd3e3d-goog
Subject: [kvm-unit-tests PATCH v3 5/6] x86: use a non-negative number in shift
From: Bill Wendling <morbo@google.com>
To: kvm@vger.kernel.org, pbonzini@redhat.com, thuth@redhat.com
Cc: jmattson@google.com, sean.j.christopherson@intel.com,
        Bill Wendling <morbo@google.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

Shifting a negative number is undefined. Clang complains about it:

x86/svm.c:1131:38: error: shifting a negative signed value is undefined [-Werror,-Wshift-negative-value]
    test->vmcb->control.tsc_offset = TSC_OFFSET_VALUE;

Using "~0ull" results in identical asm code:

	before: movabsq $-281474976710656, %rsi
	after:  movabsq $-281474976710656, %rsi

Signed-off-by: Bill Wendling <morbo@google.com>
---
 x86/svm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x86/svm.c b/x86/svm.c
index 4ddfaa4..cef43d5 100644
--- a/x86/svm.c
+++ b/x86/svm.c
@@ -1122,7 +1122,7 @@ static bool npt_rw_l1mmio_check(struct test *test)
 }
 
 #define TSC_ADJUST_VALUE    (1ll << 32)
-#define TSC_OFFSET_VALUE    (-1ll << 48)
+#define TSC_OFFSET_VALUE    (~0ull << 48)
 static bool ok;
 
 static void tsc_adjust_prepare(struct test *test)

From patchwork Wed Oct 30 21:04:19 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Bill Wendling <morbo@google.com>
X-Patchwork-Id: 11220069
Return-Path: <SRS0=Bujf=YX=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id A3239913
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 30 Oct 2019 21:04:44 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id 7D88020862
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 30 Oct 2019 21:04:44 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (2048-bit key) header.d=google.com header.i=@google.com
 header.b="YKOji3Vt"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727110AbfJ3VEn (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 30 Oct 2019 17:04:43 -0400
Received: from mail-pl1-f202.google.com ([209.85.214.202]:53174 "EHLO
        mail-pl1-f202.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1727129AbfJ3VEn (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 30 Oct 2019 17:04:43 -0400
Received: by mail-pl1-f202.google.com with SMTP id g4so1180108plj.19
        for <kvm@vger.kernel.org>; Wed, 30 Oct 2019 14:04:42 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=date:in-reply-to:message-id:mime-version:references:subject:from:to
         :cc;
        bh=lhVsM4FQB61bpLTEu+wM1raoKVMWWZdSz94pZzLGKoc=;
        b=YKOji3VtEg9GP4M9W1hLFkEHzud1IjevHtTgkkJb/AF8GltzeAdn8mWyLPrGRQUDnJ
         U1kVxLt9mOaAOSF56zDNfB47cj1PdtiXstQ/hcWILuYcG5qbAZao7O3KHOR+a2okvSnM
         JOaj1hdMepa+pzkpLtRgIji7IjdrsHxxVE7Dw2RoLzRiI5uB5t0Oh820oMiTNWj5hMcf
         8LKQVDd00M0g9y+1xAzlvmuhnP14tWAvbKxC19x2NDjdG4D68snzJNQCAOYI9ACdZxN6
         zD6fOcxV4u/fVvjmcGM9MwJggaVFNY6ye8AnzMj1kHt0bNmkTjq5Y7djC6WDs86IhjOs
         jUNA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:date:in-reply-to:message-id:mime-version
         :references:subject:from:to:cc;
        bh=lhVsM4FQB61bpLTEu+wM1raoKVMWWZdSz94pZzLGKoc=;
        b=rqLthGREVDWaG+Y8RInKrTAA8Mxfxw8lj8eHoXqjIEYxHep98weZtpsEbdoNs+068C
         dKCrI0ghPAbCNeJofY1awc6XlpXFqUZmuyJAh8w/Tug4oOh8/rexBRPKkxRS7oVeveQx
         3cyEbWTOP5DJ31t0mTZ4x68q/H2Rl5i7f3JeU7KXXsJmPmGnXAg6MuIHRUhNlewRqo14
         Oz7NHMDKHS7oJxKojDatTZQzhVaYQadAYq+yYPDg9lPCt96N3QkxF9bbJbO63OnKRdPn
         2c/7atbbtl/0l8FbG7T51S952xaAfTgmyHlkrlXCjtrC1CDcuEpXOkxDPK8s8ccrWSRz
         vTCA==
X-Gm-Message-State: APjAAAUsVdW2Hz5pICc4505y5QHzSGK7XxzpPySfEf7jEcn8JgA7DhHn
        jKSxweEJY4BOwAPDhuzSPLbPix2uk/gLdVYcKIj/ioHkVoGl3m7yQkFFQbRuDxHrRij6Casw9SI
        7G9zVgHrh6UFudHwV/+t/eJjXlPekxfFjIxRHACPzSWWX2N+mHaCyRg==
X-Google-Smtp-Source: 
 APXvYqwJ4A+cwb37fiyKOWBi6v/anubjM6fhdkdMJyrSm+NE1glPYlPeupMFqsLwX1YPK0B8/iJdtKJf2A==
X-Received: by 2002:a63:6744:: with SMTP id b65mr1593294pgc.13.1572469482038;
 Wed, 30 Oct 2019 14:04:42 -0700 (PDT)
Date: Wed, 30 Oct 2019 14:04:19 -0700
In-Reply-To: <20191030210419.213407-1-morbo@google.com>
Message-Id: <20191030210419.213407-7-morbo@google.com>
Mime-Version: 1.0
References: <20191015000411.59740-1-morbo@google.com>
 <20191030210419.213407-1-morbo@google.com>
X-Mailer: git-send-email 2.24.0.rc1.363.gb1bccd3e3d-goog
Subject: [kvm-unit-tests PATCH v3 6/6] x86: use inline asm to retrieve stack
 pointer
From: Bill Wendling <morbo@google.com>
To: kvm@vger.kernel.org, pbonzini@redhat.com, thuth@redhat.com
Cc: jmattson@google.com, sean.j.christopherson@intel.com,
        Bill Wendling <morbo@google.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

According to GCC's documentation, the only supported use for specifying
registers for local variables is "to specify registers for input and
output operands when calling Extended asm." Using it as a shortcut to
get the value in a register isn't guaranteed to work, and clang
complains that the variable is uninitialized.

Signed-off-by: Bill Wendling <morbo@google.com>
---
 x86/vmx_tests.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/x86/vmx_tests.c b/x86/vmx_tests.c
index 4aebc3f..565b69b 100644
--- a/x86/vmx_tests.c
+++ b/x86/vmx_tests.c
@@ -2138,7 +2138,9 @@ static void into_guest_main(void)
 		.offset = (uintptr_t)&&into,
 		.selector = KERNEL_CS32,
 	};
-	register uintptr_t rsp asm("rsp");
+	uintptr_t rsp;
+
+	asm volatile ("mov %%rsp, %0" : "=r"(rsp));
 
 	if (fp.offset != (uintptr_t)&&into) {
 		printf("Code address too high.\n");
@@ -3221,7 +3223,9 @@ static void try_compat_invvpid(void *unused)
 		.offset = (uintptr_t)&&invvpid,
 		.selector = KERNEL_CS32,
 	};
-	register uintptr_t rsp asm("rsp");
+	uintptr_t rsp;
+
+	asm volatile ("mov %%rsp, %0" : "=r"(rsp));
 
 	TEST_ASSERT_MSG(fp.offset == (uintptr_t)&&invvpid,
 			"Code address too high.");
